### PR TITLE
Remove launcherRunsWorkload flag

### LINKED ISSUE #329: [feature] Remove launcherRunsWorkload flag
Now we have a flag to determine that if we should use GPU in the launcher. 

But it is not expected in our situation. If we mount a GPU card to the launcher but the flag is disabled, we cannot see this card because we will set NVIDIA_VISIBLE_DEVICES to empty and libnvidia-container will not mount drivers to the pod.

We can remove the flag and rely on the launcher.resources.gpu to determine it.

### PR DESCRIPTION
**Pull request by Garrybest**
fix #329

### CODE REVIEW

File: pkg/controllers/v1/mpi_job_controller.go [hunk 1]
```diff
@@ -1476,6 +1471,7 @@ func isCleanUpPods(cleanPodPolicy *common.CleanPodPolicy) bool {
 	return false
 }
 
+// isGPULauncher check whether launcher needs GPU.
```
Comment (gaocegege) [inline pkg/controllers/v1/mpi_job_controller.go:1474]:
```suggestion
// isGPULauncher checks whether the launcher needs GPU.
```

**Review by gaocegege — COMMENTED** [review]
Generally LGTM

Thanks for your contribution! :tada: :+1:

**Comment by gaocegege** [comment]
Please sign the CLA

**Comment by Garrybest** [comment]
@googlebot I fixed it.

**Comment by Garrybest** [comment]
@googlebot I fixed it.

File: pkg/controllers/v1/mpi_job_controller.go (hunk 1 above)
Comment (Garrybest) [reply pkg/controllers/v1/mpi_job_controller.go]: Thanks for reviewing. 😄

**Review by terrytangyuan — COMMENTED** [review]
/cc @allenzfan @carmark

**Review by xieydd — APPROVED** [review]

**Review by carmark — APPROVED** [review]

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**